airflow/models.py

```diff
@@ -3455,14 +3366,176 @@ def find(dag_id, run_id=None, state=None, external_trigger=None, session=None):
         qry = session.query(DR).filter(DR.dag_id == dag_id)
         if run_id:
             qry = qry.filter(DR.run_id == run_id)
+        if execution_date:
+            qry = qry.filter(DR.execution_date == execution_date)
         if state:
             qry = qry.filter(DR.state == state)
         if external_trigger:
             qry = qry.filter(DR.external_trigger == external_trigger)
-        dr = qry.all()
+        dr = qry.order_by(DR.execution_date).all()
```

Comment:
Sorting is faster done by the db. I could add a sort by parameter. But overhead is very small as it will be on the index